static/app/utils/performanceForSentry.tsx

```diff
@@ -29,48 +30,109 @@ export function onRenderCallback(
   }
 }
 
-export const VisuallyCompleteWithData = ({
-  id,
-  hasData,
-  children,
-}: {
-  children: ReactNode;
-  hasData: boolean;
-  id: string;
-}) => {
-  const isVisuallyCompleteSet = useRef(false);
-  const isDataCompleteSet = useRef(false);
-  const longTaskCount = useRef(0);
-
-  useEffect(() => {
-    let observer;
+class LongTaskObserver {
+  private static observer: PerformanceObserver;
+  private static longTaskCount = 0;
+  private static currentId: string;
+  static getPerformanceObserver(id: string): PerformanceObserver | null {
     try {
+      LongTaskObserver.currentId = id;
+      if (LongTaskObserver.observer) {
```

Comment:
@k-fish how will a new `PerformanceObserver` get created again once it's disconnected in a `useEffect` hook?